src/presence/__tests__/heartbeat-test.js

```diff
@@ -0,0 +1,162 @@
+// @flow strict-local
+// import type { JestMockFn } from 'jest';
+import Heartbeat from '../heartbeat';
+
+// type alias for Jest callback functions of type (boolean) => void
+type CallbackType = JestMockFn<$ReadOnlyArray<boolean>, void>;
+
+describe('Heartbeat', () => {
+  // arbitrarily, one full hour between heartbeats
+  const HEARTBEAT_TIME = 60 * 60 * 1000;
+
+  // before running tests: set up fake timer API
+  beforeAll(() => {
+    jest.useFakeTimers();
+  });
+
+  // before each test: reset fake-timers state
+  beforeEach(() => {
+    jest.useFakeTimers();
```

Comment:
Do we want `useFakeTimers` in both of these, or just one?

Reply:
Just the first one. Thanks for the catch.